sycl/test-e2e/lit.cfg.py

```diff
@@ -45,6 +45,7 @@
 elif config.test_mode == "run-only":
     lit_config.note("run-only test mode enabled, only executing tests")
     config.available_features.add("run-mode")
+    config.available_features.add("build-and-run-mode")
```

Comment:
just looking at the if-else, adding this seems weird. is there another way to achieve the result while keeping the features accurate?

Reply:
Yep, I had my doubts about doing it this way. I think we have two options here not too sure what would be preferred.

1 - Introduce a testing-mode that does what this pr is making `run-only` do, and keep `run-only` unchanged
2 - Change the name of run-only to reflect the fact that it can also run tests marked as exceptions by falling back to building and running
3 - remove the build-and-run-mode feature and mark exceptions with `REQUIRES: run-mode` instead.

Personally I like option 3 the best but didn't want to jump to it right away without some discussion since it would require me to change the usage of build-and-run-mode in all the tests.

Reply:
With this pr as is we will compile tests that are currently marked with `REQUIRES: build-and-run-mode` when in run-only mode. What option 3 changes would be the feature that we would use to mark the exceptions, so instead we would use `REQUIRES: run-mode`, which both run-only and full testing modes already have.

The list of tests that are marked as splitting exceptions would not change, we would just need to update how they are marked.

Also should point out the motivation for this change is so that if we are replacing one of the ci jobs that tests in full mode with split build and run, that we retain the same test coverage on the run-only side. (Working on these changes here #16321)

Reply:
i prefer option 3, not sure about @aelovikov-intel or @uditagarwal97

Reply:
IMO, don't fix this, at least don't do any long-term fixes. The reason we have `REQUIRES: build-and-run-mode` is because we started with the simplest possible implementation of this feature and to avoid any hacks/smartness. What we really need is to provide some generic enough logic to avoid these exceptions altogether.

The priority should be switching gen12 precommit to this new mode first and asap. If we have to exclude 50 or so tests from it, let it be. 12vs40 mins justify that. We'll keep full coverage in a) post-commit/nightly, b) once e2e on PVC is re-enabled (someone is working on this, right?).

Now, if we want to enable something like this for the very short-term, let it be, and it doesn't matter much how exactly because it would be temporary anyway. I'd personally introduce a separate option like `--param prefer_build_in_run_mode_over_skip=True` to do that.

Reply:
Ok i've enabled this through a parameter instead.